Polly/src/actions/auth.js

```diff
@@ -17,27 +17,45 @@ export const checkAuth = () => {
       }
     } catch (error) {
       return dispatch({
-        type: SET_ACCESS_TOKEN_FAILURE,
+        type: SET_JSON_WEB_TOKEN_FAILURE,
         error
       })
     }
   }
 }
 
-export const setAccessToken = accessToken => {
+export const setJSONWebToken = JSONWebToken => {
   return async dispatch => {
     try {
-      await AsyncStorage.setItem("accessToken", accessToken)
+      await AsyncStorage.setItem("JSONWebToken", JSONWebToken)
       dispatch({
-        type: SET_ACCESS_TOKEN,
-        accessToken
+        type: SET_JSON_WEB_TOKEN,
+        JSONWebToken
       })
     } catch (error) {
       dispatch({
-        type: SET_ACCESS_TOKEN_FAILURE,
+        type: SET_JSON_WEB_TOKEN_FAILURE,
         error
       })
     }
   }
 }
 
+export const createNewUserRequest = async (accessToken, props) => {
+  try {
+    let response = await fetch('http://localhost:5000/user/', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          "accessToken": accessToken,
+        }),
+    })
+    let responseJson = await response.json()
+    let JSONWebToken = await responseJson.token
+    return props.setJSONWebToken(JSONWebToken)
+  } catch (error) {
+    console.log(error)
+  }
+}
```

Comment:
you shouldn't have to pass props in to this function. You should define this similar to the `setAccessToken` function, which is called an `action creator`.

Reply:
```
export const createNewUserRequest = accessToken => {
    return async dispatch => {
      try {
        const response = await fetch('http://localhost:5000/user/', {
            method: 'POST',
            headers: {
              'Content-Type': 'application/json',
            },
            body: JSON.stringify({
              "accessToken": accessToken,
            }),
        })
        const responseJson = await response.json()
        const JSONWebToken = responseJson.token

        return dispatch({
          type: SET_JWT,
          token: JSONWebToken
        })
      } catch (error) {
        return dispatch({
          type: SET_JWT_ERROR,
          error
        })
      }
    }
  }
```

Reply:
something like this ^ And then in `src/scenes/LoginScreen/LoginScreen.js` you import this function and map it to a prop, which you could call `setJwt`, and then in `src/scenes/LoginScreen/FBLoginButton.js` you can just call `this.props.setJwt(accessToken)`

Reply:
you're basically replacing the setAccessToken function with your setJSONWebToken function. It should just work the same way